$OpenBSD: patch-tools_clang_lib_Basic_Targets_cpp,v 1.16 2016/09/24 12:35:38 ajacoutot Exp $

- [OpenBSD] Add type sign information for OpenBSD
- Set ABI for ARM

--- tools/clang/lib/Basic/Targets.cpp.orig	Tue May 24 15:36:54 2016
+++ tools/clang/lib/Basic/Targets.cpp	Mon Sep 19 20:59:16 2016
@@ -4275,8 +4275,10 @@ class ARMTargetInfo : public TargetInfo {
     DoubleAlign = LongLongAlign = LongDoubleAlign = SuitableAlign = 64;
     const llvm::Triple &T = getTriple();
 
-    // size_t is unsigned long on MachO-derived environments, NetBSD and Bitrig.
+    // size_t is unsigned long on MachO-derived environments, NetBSD,
+    // OpenBSD and Bitrig.
     if (T.isOSBinFormatMachO() || T.getOS() == llvm::Triple::NetBSD ||
+        T.getOS() == llvm::Triple::OpenBSD ||
         T.getOS() == llvm::Triple::Bitrig)
       SizeType = UnsignedLong;
     else
@@ -4284,6 +4286,7 @@ class ARMTargetInfo : public TargetInfo {
 
     switch (T.getOS()) {
     case llvm::Triple::NetBSD:
+    case llvm::Triple::OpenBSD:
       WCharType = SignedInt;
       break;
     case llvm::Triple::Win32:
@@ -4475,6 +4478,7 @@ class ARMTargetInfo : public TargetInfo {
 
     switch (getTriple().getOS()) {
     case llvm::Triple::NetBSD:
+    case llvm::Triple::OpenBSD:
       PtrDiffType = SignedLong;
       break;
     default:
@@ -4525,6 +4529,8 @@ class ARMTargetInfo : public TargetInfo {
       default:
         if (Triple.getOS() == llvm::Triple::NetBSD)
           setABI("apcs-gnu");
+        else if (Triple.getOS() == llvm::Triple::OpenBSD)
+          setABI("aapcs-linux");
         else
           setABI("aapcs");
         break;
